### ISSUE TITLE
GET P2P, Funding, and Pay data

### ISSUE THREAD

**Issue by laresbernardo** [datawookie/binance#5]
Hi! Is is possible to fetch an of the following information through the API?
1. P2P personal buys and sells
2. P2P quotes for buy and sell given crypto and currency
3. Funding (tab) daily balance
4. Pay movements - internal transactions to other Binance users

Maybe if you know at least the endpoint I could help implement any/some of these.
Thanks!

**Comment by laresbernardo** [comment]
- Regarding the P2P API, some guy did something similar with python [here](https://stackoverflow.com/questions/67793326/api-binance-p2p-i-only-access-a-part-only-the-buy-and-not-all-of-it-buy-and-s).
- No, actually `wallet_daily_snapshot()` returns 0 for every day (even though it detects USDT) but I do have available balance in Funding. So makes sense that there's no balance in spot but then where is it?
```
# A tibble: 53 × 7
   type  time                total_asset_of_btc asset  free locked total
   <chr> <dttm>              <chr>              <chr> <dbl>  <dbl> <dbl>
 1 spot  2021-09-07 23:59:59 0                  USDT      0      0     0
 2 spot  2021-09-08 23:59:59 0                  USDT      0      0     0
 3 spot  2021-09-09 23:59:59 0                  USDT      0      0     0
 4 spot  2021-09-10 23:59:59 0                  USDT      0      0     0
 5 spot  2021-09-11 23:59:59 0                  USDT      0      0     0
 6 spot  2021-09-12 23:59:59 0                  USDT      0      0     0
 7 spot  2021-09-13 23:59:59 0                  USDT      0      0     0
 8 spot  2021-09-14 23:59:59 0                  USDT      0      0     0
 9 spot  2021-09-15 23:59:59 0                  USDT      0      0     0
10 spot  2021-09-16 23:59:59 0                  USDT      0      0     0
# … with 43 more rows
```
- I'll check the docs to read about transfers, but don't even know where to start as it is quite extensive.

**Comment by datawookie** [comment]
> So makes sense that there's no balance in spot but then where is it?

I have balance in various currencies in my spot trading account and I see those balances in the output from `wallet_daily_snapshot()`. Are you sure that your money is in your spot account? You can also try `"SPOT"`, `"MARGIN"`, and `"FUTURES"` as the `type` argument.

**Comment by laresbernardo** [comment]
Yes, I actually have tried all those types and none return my fund balance. Take a look of this screenshot.. I passed an amount to spot, but most of it is located within Funding. This has been filled by trading via P2P and emptied through Funding tab when I pay another user directly to his user. 
![IMG_9922](https://user-images.githubusercontent.com/25909073/139776381-1dc9c772-0afc-4760-b2e1-c9159d5f29f6.jpeg)

**Comment by datawookie** [comment]
![image](https://user-images.githubusercontent.com/6134409/139790175-ee9a7ea3-ca5d-40bb-9230-7f1cfba00e64.png)

**Comment by laresbernardo** [comment]
Correct. Those 100 USDT were passed from Funding portfolio to Spot portfolio to test. But where is the “funding” daily balance available?

**Comment by datawookie** [comment]
Not sure. See if you can find the appropriate endpoint [here](https://binance-docs.github.io/apidocs/spot/en/) and I'll happily implement.

**Comment by datawookie** [comment]
@laresbernardo I've implemented some functions for accessing the P2P data. See https://datawookie.dev/blog/2021/11/binance-p2p-trades/ for details.

**Comment by laresbernardo** [comment]
Wow, you did an incredible work here @datawookie thanks! Do you know if it's possible to retrieve any kind of historical values on P2P endpoint?

**Comment by datawookie** [comment]
> Do you know if it's possible to retrieve any kind of historical values on P2P endpoint?

**Comment by datawookie** [comment]
> Should I create a new ticket with these 2 asks?

Yes, please. Just makes it easier for me to manage.

**Comment by laresbernardo** [comment]
> Can you view these data in the Binance website? If so then yes we can probably do this. Otherwise I don't think so.

No. I've never seen this data on the app/website. I guess we'd have to save it locally with no backfill information. Thnks!

**Comment by datawookie** [comment]
Okay, so let's think about how we could do this. What data would you want to have? And what frequency? Would daily suffice?

**Comment by laresbernardo** [comment]
Maybe have a param on frequency (minute, hour, day) if possible. Daily only would work too. The data would be the same as P2P current data or if we can get the "closed P2P transactions" to have a real sense on volume, prices, payment methods... no idea how to do this though.
